modules/serverpod_auth/serverpod_auth_server/lib/src/business/config.dart

```diff
@@ -34,6 +34,19 @@ typedef SendPasswordResetEmailCallback = Future<bool> Function(
 typedef SendValidationEmailCallback = Future<bool> Function(
     Session session, String email, String validationCode);
 
+/// Callback for generation of the hash password
+typedef PasswordHashGenerator = Future<String> Function(String password);
```

Comment:
Chore: add a note stating:  `Warning: Using a custom hashing algorithm for passwords will permanently disrupt compatibility with Serverpod's validation and password hash migration.`.